packages/devtools_app/lib/src/framework/html_framework.dart

```diff
@@ -586,39 +611,113 @@ class HtmlStatusItem {
 }
 
 class HtmlToast extends CoreElement {
-  HtmlToast({this.title, this.message}) : super('div', classes: 'toast') {
+  HtmlToast({this.title, @required this.message})
+      : super('div', classes: 'toast') {
     if (title != null) {
       add(label(text: title));
     }
     add(div(text: message));
-  }
 
-  static const Duration animationDelay = Duration(milliseconds: 500);
-  static const Duration defaultHideDelay = Duration(seconds: 4);
-  static const Duration extendedHideDelay = Duration(seconds: 10);
+    toastAnimator = ToastAnimator(this);
+  }
 
   final String title;
-  @required
+
   final String message;
 
-  void show({Duration hideDelay = defaultHideDelay}) async {
-    await window.animationFrame;
+  ToastAnimator toastAnimator;
 
-    element.style.left = '0px';
+  void show({ToastStyle style = ToastStyle.normal}) async {
+    toastAnimator.show(style: style);
+  }
 
-    Timer(animationDelay, () {
-      Timer(hideDelay, _hide);
-    });
+  @override
+  String toString() => '$title $message';
+}
+
+class HtmlSurveyToast extends CoreElement {
+  HtmlSurveyToast(this._url) : super('div', classes: 'toast') {
+    toastAnimator = ToastAnimator(this);
+
+    layoutVertical();
+    add([
+      div()
+        ..layoutHorizontal()
+        ..add([
+          label(text: 'Help improve DevTools! Take our Q3 ', c: 'toast-title'),
+          surveyLink = a(text: 'survey', href: _url, target: '_blank'),
+          label(text: '.'),
+          div()..flex(),
+          span(c: 'octicon octicon-x flash-close js-flash-close')
+            ..click(() {
+              toastAnimator.hide();
+            }),
+        ]),
+      div(
+          text:
+              'By clicking on this link, you agree to share feature usage along'
+              ' with the survey responses.'),
+    ]);
   }
 
-  void _hide() {
-    element.style.left = '400px';
+  String get url => _url;
+  String _url;
+  set url(String url) {
+    _url = url;
+    surveyLink.setAttribute('href', url);
+  }
 
-    Timer(animationDelay, dispose);
+  CoreElement surveyLink;
+
+  ToastAnimator toastAnimator;
+
+  void show() async {
+    toastAnimator.show(style: ToastStyle.neverHide);
   }
+}
 
-  @override
-  String toString() => '$title $message';
+class ToastAnimator {
+  ToastAnimator(this.element);
+
+  static const Duration _animationDelay = Duration(milliseconds: 500);
+
+  static const Duration _defaultHideDelay = Duration(seconds: 4);
+
+  static const Duration _extendedHideDelay = Duration(seconds: 10);
+
+  final CoreElement element;
+
+  void show({style = ToastStyle.normal}) async {
+    await window.animationFrame;
+
+    element.element.style.left = '0px';
+
+    switch (style) {
+      case ToastStyle.normal:
```

Comment:
nit: if the style is just a duration call it a duration.
You could have
ToastDuration.default. long, and infinite.

Reply:
actually lets use duration explicitly. you could define constants for what we consider a default, extended, and infinite duration. 
Example of related Flutter api:
https://stackoverflow.com/questions/52483649/snackbar-duration-and-height

Reply:
done.